src/cogs/help.py

```diff
@@ -0,0 +1,234 @@
+from contextlib import suppress
+
+import discord
+from discord.ext import commands
+
+
+def find_ci(dictionary, string):
+    """
+    Returns a value for a key in a dictionary. Unlike normal indexing
+    this is case insensitive, and returns None if no value exists
+    """
+    for key, value in dictionary.items():
+        if key.lower() == string.lower():
+            return value
+    return None
+
+
+def cog_values(bot):
+    """Returns json for all cogs, commands and their related info."""
+    disabled_cmds = []
+    enabled_cmds = []
+    cog_dict = {}
+    misc_cmd_dict = {}
+    all_cmds = {}
+
+    for cog in bot.cogs:
+        current_cog_disabled_cmds = []
+        cmd_dict = {}
+
+        for cmd in bot.get_cog(cog).get_commands():
+            cmd_info = {
+                "name": cmd.name,
+                "usage": cmd.usage,
+                "doc_string": cmd.help,
+                "aliases": cmd.aliases,
+            }
+            cmd_item = {cmd.name: cmd_info}
+            cmd_dict[cmd.name] = cmd_info
+            all_cmds[cmd.name] = cmd_info
+
+            cmd_dict.update(cmd_item)
+            all_cmds.update(cmd_item)
+            if not cmd.enabled:
+                disabled_cmds.append(cmd.name)
+                current_cog_disabled_cmds.append(cmd.name)
+            else:
+                enabled_cmds.append(cmd.name)
+
+        cmd_count = len(bot.get_cog(cog).get_commands()) - len(
+            current_cog_disabled_cmds
+        )
+
+        cog_dict[cog] = {
+            "command_count": str(cmd_count),
+            "commands": cmd_dict,
+            "disabled_commands": current_cog_disabled_cmds,
+        }
+
+        cog_dict["disabled_commands"] = disabled_cmds
+        cog_dict["enabled_commands"] = enabled_cmds
+
+    for cmd in bot.walk_commands():
+        if not cmd.name in cmd_dict.keys():
+            misc_cmd_dict[cmd.name] = {
+                "name": cmd.name,
+                "usage": cmd.usage,
+                "doc_string": cmd.help,
+                "aliases": cmd.aliases,
+            }
+
+    misc_commands = {"misc_commands": misc_cmd_dict}
+    all_commands = {"all_commands": all_cmds}
+    cog_dict.update(misc_commands)
+    cog_dict.update(all_commands)
+    return cog_dict
+
+
+def misc_command_str(cog_dict):
+    """Builds the uncategorized command string for the discord embed description."""
+    uncategorized_commands = ""
+    for cmd in cog_dict["misc_commands"]:
+        uncategorized_commands += f"`{cmd}`, \n"
+    return uncategorized_commands[:-3]
+
+
+def description_string_builder(prefix, cog_dict, argument):
+    """Builds the description command string for the discord embed description."""
+    parsable_json = find_ci(cog_dict, argument)
+
+    description_string = ""
+    for item in parsable_json["commands"]:  # builds the embed description
+        doc_item = parsable_json["commands"][item]
+        if doc_item["name"] in cog_dict["disabled_commands"]:
+            pass
```

Comment:
Instead of making this pass, make it continue, and then remove the else. One less indentation level on your code.